aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/CreateHandler.java

```diff
@@ -58,23 +55,30 @@ private void prepareResourceModel(final ResourceHandlerRequest<ResourceModel> re
         if (request.getDesiredResourceState() == null) {
             request.setDesiredResourceState(new ResourceModel());
         }
-
         final ResourceModel model = request.getDesiredResourceState();
-        final String identifierPrefix = request.getLogicalResourceIdentifier() == null ?
-            DEFAULT_LOG_GROUP_NAME_PREFIX :
-            request.getLogicalResourceIdentifier();
 
         if (StringUtils.isNullOrEmpty(model.getLogGroupName())) {
-            model.setLogGroupName(
-                IdentifierUtils.generateResourceIdentifier(
-                    identifierPrefix,
-                    request.getClientRequestToken(),
-                    MAX_LENGTH_LOG_GROUP_NAME
-                )
-            );
+            model.setLogGroupName(generateName(request));
         }
     }
 
+    private String generateName(final ResourceHandlerRequest<ResourceModel> request) {
+        final StringBuilder identifierPrefix = new StringBuilder();
+        // the prefix will be <stack-name>-<resource type>
```

Comment:
Could you also fix the comments, the prefix is not <resource type> instead it's resource logical ID.

Reply:
no in this case prefix is the stack name right ?

Reply:
` <stack-name>-<resource type>` ->  `<stack-name>-<resource logical ID>`

Reply:
removed unwanted comment as code is explanatory

Reply:
done